profiler/BaseProfiler.F90

```diff
@@ -100,10 +100,17 @@ subroutine start_self(this, unusable, rc)
       class(KeywordEnforcer), optional, intent(in) :: unusable
       integer, optional, intent(out) :: rc
 
+      logical :: empty_stack = .true.
+
+      !$omp master
       if (this%stack%size()/= 0) this%status = INCORRECTLY_NESTED_METERS
-      _ASSERT_RC(this%stack%size()== 0,"Timer "//this%root_node%get_name()// " is not a fresh self start",INCORRECTLY_NESTED_METERS)
+      empty_stack = this%stack%size()== 0
```

Comment:
I'm a little wary that `empty_stack` is a saved variable.   A client would have to do something fairly unusual for this to become an issue, but I'd be happier to have a line _before_ the master section that sets it to `.true.` in each call.

Reply:
Fixed as suggested.